sdk/v3/role_assignments.go

```diff
@@ -1,34 +1,66 @@
-package authz
+package sdk
 
 import (
 	"context"
 	"encoding/json"
 	"net/http"
 
 	rm "github.com/brigadecore/brigade/sdk/v3/internal/restmachinery"
-	libAuthz "github.com/brigadecore/brigade/sdk/v3/lib/authz"
 	"github.com/brigadecore/brigade/sdk/v3/meta"
 	"github.com/brigadecore/brigade/sdk/v3/restmachinery"
 )
 
 const (
+	// RoleAssignmentKind represents the canonical RoleAssignment kind string
+	RoleAssignmentKind = "RoleAssignment"
+
 	// RoleAssignmentListKind represents the canonical RoleAssignmentList kind
 	// string
 	RoleAssignmentListKind = "RoleAssignmentList"
 
 	// PrincipalTypeServiceAccount represents a principal that is a
 	// ServiceAccount.
-	PrincipalTypeServiceAccount libAuthz.PrincipalType = "SERVICE_ACCOUNT"
+	PrincipalTypeServiceAccount PrincipalType = "SERVICE_ACCOUNT"
 	// PrincipalTypeUser represents a principal that is a User.
-	PrincipalTypeUser libAuthz.PrincipalType = "USER"
+	PrincipalTypeUser PrincipalType = "USER"
 )
 
+// RoleAssignment represents the assignment of a Role to a principal such as a
+// User or ServiceAccount.
+type RoleAssignment struct {
+	// Role assigns a Role to the specified principal.
+	Role Role `json:"role"`
+	// Principal specifies the principal to whom the Role is assigned.
+	Principal PrincipalReference `json:"principal"`
+	// Scope qualifies the scope of the Role. The value is opaque and has meaning
+	// only in relation to a specific Role.
+	Scope string `json:"scope,omitempty"`
+}
+
+// MarshalJSON amends RoleAssignment instances with type metadata so that
+// clients do not need to be concerned with the tedium of doing so.
+func (r RoleAssignment) MarshalJSON() ([]byte, error) {
+	type Alias RoleAssignment
+	return json.Marshal(
+		struct {
+			meta.TypeMeta `json:",inline"`
+			Alias         `json:",inline"`
+		}{
+			TypeMeta: meta.TypeMeta{
+				APIVersion: meta.APIVersion,
+				Kind:       RoleAssignmentKind,
+			},
+			Alias: (Alias)(r),
+		},
+	)
+}
+
```

Comment:
Despite appearances in the diff, this is not new code. When packages were combined, code from two different `roel_assignments.go` go files got merged into one file. This happens in other places as well.

There's nothing net new in this whole PR.